app/transactions/admin/enrollments/create_transaction.rb

```diff
@@ -0,0 +1,161 @@
+# frozen_string_literal: true
+
+module Admin
+  module Enrollments
+    # Create enrolment after a dateTimes validation
+    class CreateTransaction # rubocop:disable Metrics/ClassLength
+      include Dry::Transaction
+
+      around :rollback_if_failure
+
+      step :check_member_if_member_is_already_enrolled
+      step :check_if_the_standard_mission_is_not_full
+      step :check_if_the_duration_is_positive
+      step :check_if_datetimes_of_enrollment_are_inside_the_mission_s_period
+      step :check_if_enrollment_are_matching_the_mission_s_timeslots
+      step :check_slot_availability_for_regulated_mission
+      step :check_cash_register_mastery
+      step :create_enrollment
+
+      private
+
+      def rollback_if_failure(input, &block)
+        result = nil
+
+        Enrollment.transaction do
+          result = block.call(Success(input))
+          raise ActiveRecord::Rollback if result.failure?
+
+          result
+        end
+        result
+      end
+
+      def check_member_if_member_is_already_enrolled(input)
+        mission = input.mission
+        member = input.member
+        failure_message = I18n.t('activerecord.errors.models.enrollment.member_already_enrolled')
```

Comment:
If you need a namespace for the transaction object locale, you can go for
`'transactions.admin.enrollments.create.the_key_that_you_want'` (just follow the folder structure basically)

You could use activerecord locale namespace if you were using these keys elsewhere ; e.g. for model validation feeback